valhalla/baldr/graphconstants.h

```diff
@@ -198,7 +198,8 @@ constexpr uint32_t kMaxCurvatureFactor = 15;
 constexpr uint32_t kMaxAddedTime = 255;
 
 // Elevation constants
-constexpr float kNoElevationData = 32768.0f;
+// this is the minimum we support, i.e. -500 m would result in "no elevation"
+constexpr float kNoElevationData = -500;
```

Comment:
this constant was only used for /trace*, in the TripLegBuilder to set `kNoElevationData` values and in the serializer. now we also use it to serialize `EdgeInfo` (s. above)

Reply:
The previous value is the no data value in srtm, which is the data set we use to add eleveation. I'm certain we also reference this value in skadi just not this constant. What this means is that it's possible skadi returns the previous value when it has no data. I'll have to go check on that when I get a chance or maybe you already looked?

Reply:
I _think_ actually the tiles' nodata value is the negative equivalent, which seems to be acknowledged in skadi as well: 
https://github.com/valhalla/valhalla/blob/master/src/skadi/sample.cc#L33

that constant was just used for communication between triplegbuilder and the serializer. there's no other use of `32768` in the code base, so I assume this is fine to go with. I'll re-think (and test) a little more though, understand the relationship of all these "no data" values. would be nice to consolidate a little.

Reply:
I looked into the details and all things accessing elevation. that previous value was sorta arbitrary it seems. in skadi it's -32768.0f.

Reply:
yeah the value in skadi is the NO_DATA value in SRTM which is what the raw data will return if you ask it for a location where it doesnt have data. when we store it in the tiles we have to discretize the data further to pack it into a smaller number of bits. because of this we set a min of -500 and a max of however many descrete values we can have based on how many bits wide the field is. so that makes sense to me. default value should be no data which in this case would be -500, the constant in edgeinfo that you moved over to the graph constants. so that looks all good. the only question i have after this is when we are building tiles do we check for the skadi no data value and then convert to our -500 no data value when we store it in the tiles?

Reply:
ah got ya.. probably not! will take a look

Reply:
nope, it doesn't..  it looks to me like it never did actually. when building elevation for the tiles, it gets all height samples including skadi's `NO_DATA_VALUE`, then when computing the `weighted_grade` it just skips any `NO_DATA_VALUE` which would result in 0 elevation if there's no elevation data in the tiles, which then gets set on the tiles. so in the elevation builder we have to check for that and set the -500. I'll commit quickly

Reply:
jep got it @kevinkreiser . I changed the logic slightly in `skadi/util.cc:weighted_grade`